dev-libs/boost/patches/boost-1.69.0.patchset

```diff
@@ -174,5 +174,34 @@ index 7e46e38..67293a0 100755
          BOOST_JAM_CC=gcc
          esac
 -- 
-2.19.1
+2.37.3
+
+
+From c102b72718638d5a02ee20c89c515a6b2aedeac5 Mon Sep 17 00:00:00 2001
+From: begasus <begasus@gmail.com>
+Date: Sun, 18 Dec 2022 08:16:06 +0100
+Subject: fix build, kudos to Jessicah
+
+
+diff --git a/libs/filesystem/src/operations.cpp b/libs/filesystem/src/operations.cpp
+index 53dcdb7..6809d2d 100644
+--- a/libs/filesystem/src/operations.cpp
++++ b/libs/filesystem/src/operations.cpp
+@@ -2112,8 +2112,12 @@ namespace
+     error_code ec = path_max(path_size);
+     if (ec)return ec;
+     dirent de;
+-    buffer = std::malloc((sizeof(dirent) - sizeof(de.d_name))
+-      +  path_size + 1); // + 1 for "/0"
++	#ifdef __HAIKU__
++		buffer = std::malloc(sizeof(dirent));
```

Comment:
This is wrong and will cause buffer overflows. You need to add the `+ path_size + 1` after `sizeof(dirent)`.

Reply:
Like: `buffer = std::malloc(sizeof(dirent) + path_size + 1);` ? (checking atm)

Reply:
Added